crates/query-engine/translation/tests/tests.rs

```diff
@@ -12,6 +12,12 @@ fn select_array_column() {
     insta::assert_snapshot!(result);
 }
 
+#[test]
+fn select_array_column_nested_types() {
+    let result = common::test_translation("select_array_column_nested_types").unwrap();
+    insta::assert_snapshot!(result);
+}
```

Comment:
It's unclear to me; is this a test that would previously have failed, or is it a test that would have passed but was simply not present?

Reply:
Sorry - that is indeed not clear.

This is a test that was just not present. It passed before and after this PR.

Each of the adding of the test and the revision of the values-translation-logic could sensibly have been submitted as separate PRs.